### ISSUE TITLE
The rabbitmq consumer cannot be automatically restored after the shutdown

### ISSUE THREAD

**Issue by litiansum** [dotnetcore/CAP#1459]
We found that in an unstable network environment, when a consumer disconnects and re-subscribing times out, it cannot be automatically restored.
TimeoutExcetion for QueueBind does not appear to be handled in version 7.2.2.

![image](https://github.com/dotnetcore/CAP/assets/19256832/73565509-6b12-47dd-9ea0-70f94afd5ea5)

**Comment by yang-xiaodong** [comment]
Yes, this line `using (var client = _consumerClientFactory.Create(matchGroup.Key))`

**Comment by litiansum** [comment]
I find that this is the line that throws a TimeoutException:

`client.Subscribe(topicIds);`

Is it possible to add a TimeoutException exception handling here? For example:
![image](https://github.com/dotnetcore/CAP/assets/19256832/180c9ddb-bcf5-4023-9f3d-c32b76c4ee73)

**Comment by yang-xiaodong** [comment]
No,
1. Here is the abstraction of all Brokers. Not all Brokers will issue TimeoutException.
2. Not all TimeoutExceptions are considered unhealthy and require consumer to restart.
3. TimeoutException has been detected in `ClientFactory.Create`, indicating that your network processing is extremely unstable.

**Comment by litiansum** [comment]
> No,
> 
> 1. Here is the abstraction of all Brokers. Not all Brokers will issue TimeoutException.
> 2. Not all TimeoutExceptions are considered unhealthy and require consumer to restart.
> 3. TimeoutException has been detected in , indicating that your network env is extremely unstable.`ClientFactory.Create`

Ok, our network environment is indeed very unstable for a period of time, during which consumers are easily disconnected and cannot be recovered. For this situation, do you have any better suggestions to deal with this situation?

**Comment by yang-xiaodong** [comment]
Check your network env problem, and solve it. If you cannot solve it, your overall business data consistency is faced with serious problems, not just messages, including the consistency of database, cache, etc.

**Comment by litiansum** [comment]
Our two services are currently operating completely independently. Due to anomalies in the reconnection of message queue consumers, a large amount of data has accumulated in the message queue, affecting data consistency. The on-site network is complex, and resolving network issues in a short period is challenging. I will consider alternative solutions. Thank you for your response.
